contracts/Removal.sol

```diff
@@ -33,18 +35,45 @@ contract Removal is
     bool list;
   }
 
-  uint256 private _tokenIdCounter;
+  /**
+   * @notice Role conferring permission to initialize contract instance variables.
+   *
+   * @dev only the market contract should have this role.
+   */
+  bytes32 public constant CONTRACT_INITIALIZER_ROLE =
+    keccak256("CONTRACT_INITIALIZER_ROLE");
+
+  /**
+   * @notice The RestrictedNORI contract that manages restricted tokens.
+   */
+  RestrictedNORI private _restrictedNori;
+  uint256 public tokenIdCounter;
   string public name; // todo why did I add this
   mapping(uint256 => uint256) public indexToTokenId; // todo consider how we're keeping track of the number and order of ids, ability to iterate
   mapping(uint256 => bool) private _tokenIdExists;
+  mapping(uint256 => uint256) private _idToRestrictionScheduleStartTime;
 
-  function initialize() public virtual initializer {
+  function initialize() external virtual initializer {
     super.initialize("https://nori.com/api/removal/{id}.json");
     __ERC1155Supply_init_unchained();
-    _tokenIdCounter = 0;
+    tokenIdCounter = 0;
     name = "Removal";
   }
 
+  function addContractInitializer(address _contractInitializer) public {
```

Comment:
This seems like a new pattern for setting up roles.  I'd like to understand more about why the approach is different here.

Reply:
changed